src/Workspaces/CoreTestUtilities/Remote/InProcRemoteHostClientProvider.cs

```diff
@@ -29,15 +30,37 @@ public IWorkspaceService CreateService(HostWorkspaceServices workspaceServices)
                 => new InProcRemoteHostClientProvider(workspaceServices);
         }
 
+        private sealed class WorkspaceManager : RemoteWorkspaceManager
+        {
+            private readonly Lazy<RemoteWorkspace> _lazyWorkspace;
+
+            public WorkspaceManager(Type[]? additionalRemoteParts)
+            {
+                _lazyWorkspace = new Lazy<RemoteWorkspace>(
+                    () => new RemoteWorkspace(FeaturesTestCompositions.RemoteHost.AddParts(additionalRemoteParts).GetHostServices(), WorkspaceKind.RemoteWorkspace));
+            }
+
+            public override RemoteWorkspace GetWorkspace()
+                => _lazyWorkspace.Value;
+        }
+
         private readonly HostWorkspaceServices _services;
         private readonly AsyncLazy<RemoteHostClient> _lazyClient;
 
+        public AssetStorage? RemoteAssetStorage { get; set; }
+        public Type[]? AdditionalRemoteParts { get; set; }
```

Comment:
mutable?

Reply:
You mean the array or the property?

Reply:
The properties are mutable so that test can configure the values before the client is created. 
The array can be immutable.

Reply:
ah, this is test code. that's ok then :)